src/trio/_core/_tests/test_run.py

```diff
@@ -940,10 +942,14 @@ async def main() -> None:
         _core.spawn_system_task(system_task)
         await sleep_forever()
 
+    # not wrapped in ExceptionGroup
     with pytest.raises(_core.TrioInternalError) as excinfo:
         _core.run(main)
 
-    assert RaisesGroup(KeyError, ValueError).matches(excinfo.value.__cause__)
+    # TODO: triple-wrapped exceptions ?!?!
+    assert RaisesGroup(RaisesGroup(RaisesGroup(KeyError, ValueError))).matches(
+        excinfo.value.__cause__
+    )
```

Comment:
The reason for the triple-wrapping turns out to be quite straightforward:
https://github.com/python-trio/trio/blob/aadd1ea56f2b64d920a70897abb1a7178b54cfab/src/trio/_core/_run.py#L1903-L1913

doesn't make for the cleanest `__cause__`s on `TrioInternalError`s, but don't think it's a problem. Updating comments.